### PR TITLE
Minor changes that improve performance

### PR DESCRIPTION
**Pull request by hkaiser**
- flyby: fix minor inconsistency while thread stealing
- flyby: fixing warnings on Windows

This separates some minor improvements from #5467

### CODE REVIEW

File: libs/full/runtime_configuration/src/runtime_configuration.cpp [hunk 1]
```diff
@@ -1299,7 +1274,7 @@ namespace hpx { namespace util {
             return huge_stacksize;
 
         case threads::thread_stacksize::nostack:
-            return (std::numeric_limits<std::ptrdiff_t>::max)();
+            return (std::numeric_limits<std::ptrdiff_t>::max) ();
```
Comment (msimberg) [inline libs/full/runtime_configuration/src/runtime_configuration.cpp:1277]: This needs to be reverted.

**Review by msimberg — CHANGES_REQUESTED** [review]
Nice, thanks @hkaiser! Looks good to me except the one formatting change that should be reverted.

Is that due clang-format 12 this time...? Or manually applied?

**Comment by hkaiser** [comment]
> Is that due clang-format 12 this time...? Or manually applied?

Yep, it's clang-format v12. I'll revert this.

File: libs/full/runtime_configuration/src/runtime_configuration.cpp [hunk 2]
```diff
@@ -753,40 +754,34 @@ namespace hpx { namespace util {
 
     std::size_t runtime_configuration::get_ipc_data_buffer_cache_size() const
     {
-        if (has_section("hpx.parcel"))
+        util::section const* sec = get_section("hpx.parcel.ipc");
+        if (nullptr != sec)
```
Comment (msimberg) [inline libs/full/runtime_configuration/src/runtime_configuration.cpp:758]: Purely subjective, and I don't mind to keep it like this, but what about:
```suggestion
        if (util::section const* sec = get_section("hpx.parcel.ipc")
```
If you prefer this I don't mind doing the change.

**Review by aurianer — DISMISSED** [review]
LGTM thanks!

File: libs/full/runtime_configuration/src/runtime_configuration.cpp (hunk 2 above)
Comment (hkaiser) [reply libs/full/runtime_configuration/src/runtime_configuration.cpp:758]: I can do that.

**Comment by msimberg** [comment]
@hkaiser same thing here... I've only applied a formatting change on the one commit. I think this should be ready to be merged unless any surprises come up from CI.

**Comment by hkaiser** [comment]
Thanks!

**Pull request merged**